app/assets/javascripts/application.js

```diff
@@ -463,3 +535,18 @@ function clearError(field) {
     .children('.error-message');
   error_block.remove();
 }
+
+// jQuery deprecated functions
+// used by gridster and bootstrap
+$.fn.isFunction = function(func) {
+  return typeof func === 'function';
+};
+$.fn.isArray = Array.isArray;
+$.fn.trim = String.prototype.trim;
+$.fn.bind = function(event, func) {
+  return this.on(event, func);
+};
+// used by puppet plugin
```

Comment:
Perhaps open a PR against foreman_puppet to drop this usage?

Reply:
Took a while to setup puppet, but made the pr: https://github.com/theforeman/foreman_puppet/pull/403